core/storage/topic/gae_models.py

```diff
@@ -51,6 +51,10 @@ class TopicModel(base_models.VersionedModel):
     name = ndb.StringProperty(required=True, indexed=True)
     # The canonical name of the topic, created by making `name` lowercase.
     canonical_name = ndb.StringProperty(required=True, indexed=True)
+    # The abbreviated name of the topic.
+    abbreviated_name = ndb.StringProperty(indexed=False)
+    # Topic thumbnail as a dataURI string.
+    thumbnail_data_url = ndb.TextProperty(default=None, indexed=False)
```

Comment:
Actually, I think it is better to store this in GCS, since then the image can be cached. Otherwise it needs to be reconstituted every time.

@aks681, can you suggest the filepath for where it should go, so that the image is stored separately from any other topic images?

Then we don't need this field, since we can just look up the image based on topic ID. (Sorry for not realizing this earlier.)

Reply:
Maybe, you can use /topic/thumbnails/<topicId>?
We are already using /topic/<topicId> to store the RTE images in a topic, so you can use the same functions, just with a different URL.

Reply:
@aks681 Technically this doesn't work so well if the topic ID is "thumbnails". Which generally can't happen but it feels a bit odd to structure it that way.

Are the RTE images for that topic stored in an images subfolder or in the topic root folder? If the former then making something like `/topic/<[topicId]>/thumbnails` should work, and if the latter then we could either (i) change it to the former, or (ii) just store all images in `/topic/<[topicId]>`.

Also, do you think we should be creating a folder and storing just the filename of the thumbnail image in the model? I guess that's probably better than specifying an explicit filepath since the filepath would end up fixing the format of the image (e.g. the thumbnail would always be `/topic/<[topicId]>/thumbnail.png` but we might not want to fix it as png). It would also give us the benefit of being able to keep previous thumbnail images in case we wanted to switch back to them later.

Reply:
The image is stored inside /topic/topicId/assets/image/<[url]>. Yeah, making something like /assets/thumbnails should be fine.
That's what is done now. We just store the filename of the image in the RTE content, and using the context, the full url is generated. Something similar can be done here.

Reply:
Okay, I'll make the changes to store thumbnails in `/topic/<topic-id>/assets/thumbnails`. Thanks!

Reply:
Done.